### PR TITLE
config to remove $gp, $k0, $k1 for students in cs1521

### PR DESCRIPTION
**Pull request by SpanishPear**
Welp, since I can't teach today thought I'd do this instead

### CODE REVIEW

File: crates/mipsy_web/src/components/registers.rs [hunk 1]
```diff
@@ -9,6 +9,10 @@ pub struct RegisterProps {
     pub tab: UseStateHandle<RegisterTab>,
 }
 
+const K0_INDEX: usize = 26;
```
Comment (Dylan-Brotherston) [inline crates/mipsy_web/src/components/registers.rs:12]: Can we not use register numbers from mipsy_lib?

**Review by Dylan-Brotherston — COMMENTED** [review]
I think the exact behaviour that Andrew was after was to hide any register not modified after the main: label

**Comment by SpanishPear** [comment]
> I think the exact behaviour that Andrew was after was to hide any register not modified after the main: label

In theory that's what "used registers" already does, except for the whole program, not specifically "after main".

Doing all regs modified "before main" I think would be a lot trickier, as you'd have to bind register state to `pc` and you'd have to add a level of communication between the worker and main thread

File: crates/mipsy_web/src/components/registers.rs (hunk 1 above)
Comment (SpanishPear) [reply crates/mipsy_web/src/components/registers.rs:12]: didn't even know that existed... If only I had intellisense working on windows : ^)

**Review by Dylan-Brotherston — APPROVED** [review]

**Pull request merged**